site/app/controllers/HomePageController.php

```diff
@@ -101,25 +102,64 @@ public function getGroups($user_id = null): MultiResponse {
      * @return MultiResponse
      */
     public function showHomepage() {
+        $user = $this->core->getUser();
         $courses = $this->getCourses()->json_response->json;
 
+
+
         return new MultiResponse(
             null,
             new WebResponse(
                 ['HomePage'],
                 'showHomePage',
                 $this->core->getUser(),
+                $this->core->getCsrfToken(),
                 $courses["data"]["unarchived_courses"],
                 $courses["data"]["archived_courses"]
             )
         );
     }
 
+    /**
+     * @Route("/home/get_user_time_zone", methods={"GET"})
+     * @return JsonResponse
+     *
+     * Handle ajax request to update the currently logged in user's time zone data.
+     *
+     * Will return a json success or failure response depending on the result of the operation.
+     */
+    public function getUserTimeZone() {
```

Comment:
Is there a reason why this is an AJAX request?  The system timezone could be included in the Twig logic for this page, essentially just setting the value of a JS constant.

Reply:
The AJAX request is used to retrieve the time zone that that user sets on the profile which is saved in the database.